csharp/ql/lib/semmle/code/csharp/frameworks/EntityFramework.qll

```diff
@@ -376,56 +389,142 @@ module EntityFramework {
-        c.requiresComponentStackIn(head, _, tail, _) and
-        head.getProperty() = mapped and
-        mapped = c.getAColumnProperty(_) and
-        input = SummaryComponentStack::push(SummaryComponent::content(head), tail)
+    /**
+     * Gets the string representation for a synthetic identifier for this.
+     */
+    string getSyntheticName() {
+      exists(string qualifier, string type, string name |
+        this.hasQualifiedName(qualifier, type, name)
+      |
+        result = getQualifiedName(qualifier, type, name)
       )
     }
 
-    pragma[noinline]
-    private predicate output(SummaryComponentStack output, Property mapped) {
-      exists(PropertyContent head, SummaryComponentStack tail |
-        c.requiresComponentStackOut(head, _, tail, _) and
-        head.getProperty() = mapped and
-        mapped = c.getAColumnProperty(_) and
-        output = SummaryComponentStack::push(SummaryComponent::content(head), tail)
+    /**
+     * Gets the context class where this is a Db set property.
+     */
+    DbContextClass getDbContextClass() { result = c }
+  }
+
+  /**
+   * Holds if `input` is a valid summary component stack for property `mapped`
+   * for the context class `c`.
+   */
+  pragma[noinline]
+  predicate input(DbContextClass c, SummaryComponentStack input, Property mapped) {
+    exists(PropertyContent head, SummaryComponentStack tail |
+      c.requiresComponentStackIn(head, _, tail, _) and
+      head.getProperty() = mapped and
+      mapped = c.getAColumnProperty(_) and
+      input = SummaryComponentStack::push(SummaryComponent::content(head), tail)
+    )
+  }
+
+  /**
+   * Holds if `output` is a valid summary component stack for the getter of `dbSet`
+   * for property `mapped` for the context class `c`.
+   */
+  pragma[noinline]
+  predicate output(
+    DbContextClass c, SummaryComponentStack output, Property mapped, DbContextClassSetProperty dbSet
+  ) {
+    exists(PropertyContent head, SummaryComponentStack tail |
+      c.requiresComponentStackOut(head, _, tail, _, dbSet) and
+      head.getProperty() = mapped and
+      mapped = c.getAColumnProperty(_) and
+      output = SummaryComponentStack::push(SummaryComponent::content(head), tail)
+    )
+  }
+
+  private class DbContextClassSetPropertySynthetic extends EFSummarizedCallable {
+    private DbContextClassSetProperty p;
+
+    DbContextClassSetPropertySynthetic() { this = p.getGetter() }
+
+    override predicate propagatesFlow(
+      SummaryComponentStack input, SummaryComponentStack output, boolean preservesValue
+    ) {
+      exists(SummaryComponentStack synthetic, string name, DbContextClass c, Property mapped |
+        preservesValue = true and
+        c = p.getDbContextClass() and
+        input(c, synthetic, mapped) and
+        output(c, output, mapped, p) and
+        name =
+          c.getSyntheticNames() + p.getSyntheticName() + synthetic.toString() + output.toString() and
```

Comment:
As discussed offline, I think we should use `getComponentStack` from `FlowSummaryImpl.qll` instead of `toString`

Reply:
And perhaps also a good idea to delimit the components above somehow.

Reply:
Yes, it is definitely a good idea to put in a delimiter - I will do that.
I started looking into using `getComponentStack` for the string representation, but that leads to non-monotonic recursion - so I am doing another PR as a pre-requisite for this one to avoid the non-monotonic recursion.